### PR TITLE
feat: autonomous routines, tuning, vision, oh my

### PR DESCRIPTION
**Pull request by TheGamer1002**

### CODE REVIEW

File: src/main/deploy/pathplanner/paths/Go to red speaker.path
Comment (NyxAlexandra) [inline src/main/deploy/pathplanner/paths/Go to red speaker.path:1]: `go_to_red_speaker.path` or `go-to-red-speaker.path`

File: src/main/java/frc/robot/LocalADStarAK.java [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+import org.littletonrobotics.junction.Logger;
+import org.littletonrobotics.junction.inputs.LoggableInputs;
+
+public class LocalADStarAK implements Pathfinder {
+  private final ADStarIO io = new ADStarIO();
+
+  /**
+   * Get if a new path has been calculated since the last time a path was retrieved
+   *
+   * @return True if a new path is available
```
Comment (NyxAlexandra) [inline src/main/java/frc/robot/LocalADStarAK.java:24]: redundant, remove

File: src/main/java/frc/robot/RobotContainer.java [hunk 2]
```diff
@@ -42,14 +46,14 @@ public class RobotContainer {
   private final SwerveSubsystem drivebase;
   private final Intake intake;
   private final Shooter shooter;
-  private final Command intakeCommand, shootCommand, aimAtTargetCommand, driveToShootingPositionCommand, rotateCommand;
+  private final Command intakeCommand, shootCommand, aimAtLimelightCommand, rotateCommand, driveToPoseCommand;
 
   // CommandJoystick rotationController = new CommandJoystick(1);
   // Replace with CommandPS4Controller or CommandJoystick if needed
```
Comment (NyxAlexandra) [inline src/main/java/frc/robot/RobotContainer.java:52]: remove

File: src/main/java/frc/robot/RobotContainer.java [hunk 3]
```diff
@@ -42,14 +46,14 @@ public class RobotContainer {
...
-  CommandJoystick driverController = new CommandJoystick(1);
 
   // CommandJoystick driverController   = new CommandJoystick(3);//(OperatorConstants.DRIVER_CONTROLLER_PORT);
```
Comment (NyxAlexandra) [inline src/main/java/frc/robot/RobotContainer.java:54]: remove

File: src/main/java/frc/robot/RobotContainer.java [hunk 4]
```diff
@@ -122,15 +114,16 @@ private void configureBindings() {
...
     new JoystickButton(driverXbox, 1).onTrue((new InstantCommand(drivebase::zeroGyro)));
-    new JoystickButton(driverXbox, XboxController.Button.kX.value).whileTrue(drivebase.driveCommand(() -> 0.0, () -> 0.0, () -> -LimelightHelpers.getTX("limelight-back") / 10));
-    new JoystickButton(driverXbox,
-        2).whileTrue(driveToShootingPositionCommand);
+    new JoystickButton(driverXbox, XboxController.Button.kX.value).whileTrue(aimAtLimelightCommand);
+    new JoystickButton(driverXbox, 2).whileTrue(driveToPoseCommand);
     new JoystickButton(driverXbox, XboxController.Button.kY.value).whileTrue(rotateCommand);
 //    new JoystickButton(driverXbox, 3).whileTrue(new RepeatCommand(new InstantCommand(drivebase::lock, drivebase)));
+    //new JoystickButton(driverXbox, XboxController.Button.kStart.value).whileTrue(drivebase.sysIdAngleMotorCommand());
+    //new JoystickButton(driverXbox, XboxController.Button.kBack.value).whileTrue(drivebase.sysIdDriveMotorCommand());
```
Comment (NyxAlexandra) [inline src/main/java/frc/robot/RobotContainer.java:122]: remove

File: src/main/java/frc/robot/commands/drivebase/AimAtLimelightCommand.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+   * this method returning true -- the scheduler will call its {@link #end(boolean)} method.
+   * </p><p>
+   * Returning false will result in the command never ending automatically. It may still be
+   * cancelled manually or interrupted by another command. Hard coding this command to always
+   * return true will result in the command executing once and finishing immediately. It is
+   * recommended to use * {@link edu.wpi.first.wpilibj2.command.InstantCommand InstantCommand}
+   * for such an operation.
+   * </p>
+   *
+   * @return whether this command has finished.
```
Comment (NyxAlexandra) [inline src/main/java/frc/robot/commands/drivebase/AimAtLimelightCommand.java:70]: unneeded, remove

File: src/main/java/frc/robot/subsystems/VisionSubsystem.java [hunk 6]
```diff
@@ -44,5 +44,64 @@ private VisionSubsystem() {
   public void periodic() {
     // This method will be called once per scheduler run
     //SwerveSubsystem.addVisionMeasurement(LimelightHelpers.getBotPose2d("back-limelight"), Timer.getFPGATimestamp() - LimelightHelpers.getLatency_Capture("back-limelight") - LimelightHelpers.getLatency_Pipeline("back-limelight") );
```
Comment (NyxAlexandra) [inline src/main/java/frc/robot/subsystems/VisionSubsystem.java:46]: remove

**Review by NyxAlexandra — CHANGES_REQUESTED** [review]
looks good overall. resolve the comments and run `google-java-format` to fix the extra whitespace

File: src/main/java/frc/robot/LocalADStarAK.java (hunk 1 above)
Comment (TheGamer1002) [reply src/main/java/frc/robot/LocalADStarAK.java:24]: This code is all copy-pasted, and I find this useful for when I use it, so I'm gonna keep this in

File: src/main/deploy/pathplanner/paths/Go to red speaker.path
Comment (TheGamer1002) [reply src/main/deploy/pathplanner/paths/Go to red speaker.path:1]: The file name is used in the PathPlanner auto selection GUI

**Review by NyxAlexandra — APPROVED** [review]
good enughgho

**Pull request merged**